src/test/java/jimmy/mcgymmy/model/date/DateTest.java

```diff
@@ -10,67 +10,116 @@
+    private static final String VALID_DATE_STRING_1_SLASH_D_MM_YYYY = "20/4/2020";
+    private static final String VALID_DATE_STRING_1_SLASH_YYYY_MM_DD = "2020/04/20";
+    private static final String VALID_DATE_STRING_1_SLASH_DD_MM_YYYY = "20/04/2020";
+
+    private static final String VALID_DATE_STRING_1_DASH_D_M_YYYY = "2-9-2020";
+    private static final String VALID_DATE_STRING_1_DASH_YYYY_M_D = "2020-9-2";
+    private static final String VALID_DATE_STRING_1_DASH_DD_MM_YYYY = "20-04-2020";
+    private static final String VALID_DATE_STRING_1_DASH_YYYY_DD_MM = "2020-4-20";
+    private static final String VALID_DATE_STRING_1_DASH_DD_M_YYYY = "20-4-2020";
 
+    private static final String VALID_DATE_STRING_LEAP_YEAR_FEB = "29 Feb 2020";
+    private static final String INVALID_DATE_STRING_NON_LEAP_YEAR_FEB = "29 Feb 2019";
 
     @Test
     void currentDate_returnsCorrectCurrentDate() throws IllegalValueException {
         assertEquals(Date.currentDate(), new Date(LocalDate.now().toString()));
     }
 
+    @Test
+    void leapYearValidTest() throws IllegalValueException {
+        //Valid leap year Date
+        assertEquals(new Date(VALID_DATE_STRING_LEAP_YEAR_FEB).toString(), VALID_DATE_STRING_LEAP_YEAR_FEB);
+
+        //Invalid leap year date
+        assertThrows(IllegalValueException.class, () -> new Date(INVALID_DATE_STRING_NON_LEAP_YEAR_FEB));
+
+    }
+
     @Test
     void constructorAndToString_supportedFormat_createsDateObject() throws IllegalValueException {
-        // format: "yyyy-MM-dd"
-        assertEquals(new Date("2020-04-20").toString(), EXPECTED_DATE_STRING_1);
-        // format: "yyyy-M-dd"
-        assertEquals(new Date("2020-4-20").toString(), EXPECTED_DATE_STRING_1);
+        // format: "d/M/yyyy",
+        assertEquals(new Date(VALID_DATE_STRING_1_SLASH_D_M_YYYY).toString(), VALID_DATE_STRING_2_D_MMM_YYYY);
+        // format: "d-M-yyyy"
+        assertEquals(new Date(VALID_DATE_STRING_1_DASH_D_M_YYYY).toString(), VALID_DATE_STRING_2_D_MMM_YYYY);
         // format: "yyyy-M-d"
-        assertEquals(new Date("2020-9-2").toString(), EXPECTED_DATE_STRING_2);
-        // format: "dd-MM-yyyy"
-        assertEquals(new Date("20-04-2020").toString(), EXPECTED_DATE_STRING_1);
+        assertEquals(new Date(VALID_DATE_STRING_1_DASH_YYYY_M_D).toString(), VALID_DATE_STRING_2_D_MMM_YYYY);
+        // format: "yyyy-M-dd"
+        assertEquals(new Date(VALID_DATE_STRING_1_DASH_YYYY_DD_MM).toString(), VALID_DATE_STRING_1_DD_MMM_YYYY);
         // format: "dd-M-yyyy"
-        assertEquals(new Date("20-4-2020").toString(), EXPECTED_DATE_STRING_1);
-        // format: "d-M-yyyy"
-        assertEquals(new Date("2-9-2020").toString(), EXPECTED_DATE_STRING_2);
+        assertEquals(new Date(VALID_DATE_STRING_1_DASH_DD_M_YYYY).toString(), VALID_DATE_STRING_1_DD_MMM_YYYY);
+        // format: "dd/M/yyyy"
+        assertEquals(new Date(VALID_DATE_STRING_1_SLASH_D_MM_YYYY).toString(), VALID_DATE_STRING_1_DD_MMM_YYYY);
```

Comment:
These comments do not exactly tally with the subsequent assertEquals arguments. Similar instances found below. Just a minor nit.